src/templates/Documentation.js

```diff
@@ -106,41 +110,66 @@ const ModifiedEl = styled.div`
   }
 `;
 
-const PageOutlineItem = ({ id, isActive, title }) => (
-  <NavItemEl
-    isActive={isActive}
-    onClick={(e) => {
-      e.preventDefault();
-      smoothScrollTo(document.getElementById(id));
-    }}
-  >
-    {title}
-  </NavItemEl>
-);
+const SectionEl = styled.section``;
+
+const PageOutlineItem = ({ id, isActive, title }) => {
+  const { setActiveNode, onNavClick } = React.useContext(
+    SideNavProgressContext,
+  );
+
+  return (
+    <NavItemEl
+      isActive={isActive}
+      onClick={(e) => {
+        e.preventDefault();
+        onNavClick(true);
+        setActiveNode(document.getElementById(id));
+        smoothScrollTo(document.getElementById(id), () => {
+          onNavClick(false);
```

Comment:
Where I'm toggling `onNavClick` value so we can stop scroll tracking behavior

Reply:
I think it'd be nice to keep this logic contained within the sidenav provider—since this is a page template, the the bookkeeping details of when to toggle state in the sidenav seem too granular. Not sure the tech details work out but something like `focusNavItem(id)` here would be awesome

Reply:
Not sure about putting this in the sidenav provider since both `/api` and `/docs` use different `renderItems` for its `SideNavBody`. `/api` uses `ScrollRouter` in order to change its window state when the nav is clicked where `/docs` doesn't do that. Also `smoothScrollTo` is only used in `/docs`.

Reply:
Hm maybe it's not super feasible then. This is a relatively minor note, not worth blocking the PR over